runtime/lua/vim/diagnostic.lua

```diff
@@ -430,15 +431,15 @@ local function next_diagnostic(position, search_forward, bufnr, opts, namespace)
         sort_diagnostics = function(a, b) return a.col > b.col end
         is_next = function(diagnostic) return diagnostic.col < position[2] end
       end
-      table.sort(line_diagnostics, sort_diagnostics)
+      table.sort(line_diagnostics[lnum], sort_diagnostics)
       if i == 0 then
-        for _, v in pairs(line_diagnostics) do
+        for _, v in ipairs(line_diagnostics[lnum]) do
```

Comment:
is there a reason to change to ipairs here?

Reply:
Nope, just habit, I'll switch it back.